File: src/com/train/algorithm/twoPointers/LongPressedName.cpp
//
// Created by 罗斯 on 26/8/2022.
//
#include <string>

using namespace std;
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        int i = 0,j = 0;
        while (i < name.size()&&j < typed.size()){
            if(name[i] == typed[j]){
                ++i;
                ++j;
            }else if(j > 0 && typed[j] == typed[j - 1]){
                ++j;
            }else{
                return false;
            }
        }
        while (j < typed.size() && typed[j] == typed[j - 1]) ++j;
        return j == typed.size() && i == name.size();
    }
};

File: src/com/train/algorithm/twoPointers/implementInPython/LongPressedName.py
class Solution(object):
    def isLongPressedName(self, name, typed):
        i,j = 0,0
        n,m = len(name),len(typed)

        while j < m:
            if i < n and name[i] == typed[j]:
                i += 1
            elif j == 0 or typed[j] != typed[j - 1]:
                return False
            j += 1
        return i == n